cpp/src/arrow/record_batch.h

```diff
@@ -243,6 +243,9 @@ class ARROW_EXPORT RecordBatchReader {
     return batch;
   }
 
+  /// \brief finalize reader
+  virtual Status Close() = 0;
```

Comment:
Instead of making this pure virtual (`= 0`) can you make this virtual and give it an implementation?
```suggestion
  virtual Status Close() {
    return Status::OK();
  };
```